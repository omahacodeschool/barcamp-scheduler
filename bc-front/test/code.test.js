var React = require('react')
var Title = require('../app/components/card/title.js');
var Name = require('../app/components/card/name.js');
var Description = require('../app/components/card/description.js');
var CategoryLetter = require('../app/components/card/category_letter.js');
var renderer = require('react-test-renderer');

test('Title changes to a prop value when passed', () => {
  var component = renderer.create(
    <Title title="Doing Big Things" />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Name changes to a prop value when passed', () => {
  var component = renderer.create(
    <Name name="Jamie" />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Description changes to a prop value when passed', () => {
  var component = renderer.create(
    <Description name="This is a description of something!" />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Category letter changes to a prop value when passed', () => {
  var component = renderer.create(
    <CategoryLetter name="X" />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Category image changes to a prop value when passed', () => {
  var component = renderer.create(
      <div><img src='./app/images/track_technology_2016.png' width="40" /></div>  ).toJSON();
  expect(component).toMatchSnapshot();
});