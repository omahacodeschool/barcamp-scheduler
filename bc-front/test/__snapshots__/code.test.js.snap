// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category image changes to a prop value when passed 1`] = `
<div>
  <img
    src="./app/images/track_technology_2016.png"
    width="40"
  />
</div>
`;

exports[`Category letter changes to a prop value when passed 1`] = `
<div
  style={
    Object {
      "bottom": "5px",
      "color": "navy",
      "fontFamily": "cursive",
      "fontSize": "1.75em",
      "position": "absolute",
      "right": "15px",
    }
  }
>
  N
</div>
`;

exports[`Description changes to a prop value when passed 1`] = `
<div>
  Celiac kitsch poutine, tacos stumptown cardigan jianbing pug pickled kinfolk try-hard put a bird on it air plant activated charcoal. Swag kinfolk marfa cornhole, ennui try-hard narwhal blue bottle forage gastropub schlitz keytar. Bicycle rights schlitz farm-to-table woke biodiesel, shoreditch scenester four dollar toast fashion axe heirloom godard tbh DIY skateboard. Offal cliche hashtag franzen asymmetrical art party.
</div>
`;

exports[`Name changes to a prop value when passed 1`] = `
<div
  style={
    Object {
      "float": "left",
      "fontSize": "0.75em",
      "paddingTop": "25px",
    }
  }
>
  Jamie
</div>
`;

exports[`Title changes to a prop value when passed 1`] = `
<div
  style={
    Object {
      "fontSize": "1.5em",
    }
  }
>
  Doing Big Things
</div>
`;
